service/matching/matching_engine_test.go

```diff
@@ -3547,6 +3549,91 @@ func (s *matchingEngineSuite) TestPollWorkflowTaskQueueWithRateLimiterError() {
 	s.ErrorIs(err, rateLimiterErr)
 }
 
+func (s *matchingEngineSuite) TestDispatchNexusTask_ValidateTimeoutBuffer() {
+	const ctxTimeout = 2 * time.Second
+	type testCase struct {
+		name      string
+		sleepTime time.Duration
+		assertion func(t *testing.T, response *matchingservice.DispatchNexusTaskResponse, err error)
+	}
+
+	testCases := []testCase{
+		{
+			name:      "deadline_exceeded",
+			sleepTime: ctxTimeout - NexusTimeoutBuffer,
+			assertion: func(t *testing.T, response *matchingservice.DispatchNexusTaskResponse, err error) {
+				require.NoError(t, err)
+				require.NotNil(t, response)
+				require.NotNil(t, response.Outcome)
```

Comment:
As I mentioned before, this is redundant. Applies below too. See the implementation of `GetRequestTimeout()`.